include/server/api/base_parameters_grammar.hpp

```diff
@@ -149,10 +149,17 @@ struct BaseParametersGrammar : boost::spirit::qi::grammar<Iterator, Signature>
             qi::lit("bearings=") >
             (-(qi::short_ > ',' > qi::short_))[ph::bind(add_bearing, qi::_r1, qi::_1)] % ';';
 
+        side_type.add("d", engine::Side::DEFAULT)("o", engine::Side::OPPOSITE)("b", engine::Side::BOTH);
+        sides_rule = qi::lit("sides=") >
+                        (-(side_type | qi::attr(engine::Side::BOTH)) %
```

Comment:
Here must be either `-` for `vector<optional<Side>>` or `qi::attr(engine::Side::BOTH)`  for `vector<Side>` 
`-` will place empty optionals for empty values, but `attr(BOTH)` will place `BOTH`.

`(-side_type %';')[...]` is enough for vector of optionals